File: app/src/test/java/edu/gatech/donatrix/testRegisteredUser.java
package edu.gatech.donatrix;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

import edu.gatech.donatrix.dao.Database;
import edu.gatech.donatrix.model.User;
import edu.gatech.donatrix.model.UserType;

/**
 * How to test if a user is registered.
 */
public class testRegisteredUser {
    private final Database db = Database.getInstance();
    private User good;
    private User locked;
    private User nUser;
    private User unregistered;
    private User badUser;

    /**
     * Making a few users and setting up the db
     */
    @Before
    public void setUp() {
        good = new User("zday123@yahoo.com", "janisday", "Zeke Day", false, UserType.USER);
        locked = new User("aday123@yahoo.com", "password", "Alex Day", true, UserType.USER);
        unregistered = new User("blah@blah.com", "blah", "Blah", false, UserType.USER);
        nUser = new User(null, "null", "Null", false, UserType.USER);
        badUser = new User("zday123@yahoo.com", "password", "Zeke Day", false, UserType.USER);
        db.registerUser(good);
        db.registerUser(locked);
        db.registerUser(nUser);
    }

    /**
     * Testing an unregistered user
     */
    @Test
    public void testUnregisteredUser() {
        assertFalse(db.checkRegisteredUser(unregistered.getEmail(), unregistered.getPassword()));
    }

    /**
     * Testing a locked user.
     */
    @Test
    public void testLockedUser() {
        assertFalse(db.checkRegisteredUser(locked.getEmail(), locked.getPassword()));
    }

    /**
     * Testing a null user
     */
    @Test
    public void testNullUser() {
        assertFalse(db.checkRegisteredUser(nUser.getEmail(), nUser.getPassword()));
    }

    /**
     * Test if the password is invalid.
     */
    @Test
    public void testBadPassword() {
        assertFalse(db.checkRegisteredUser(badUser.getEmail(), badUser.getPassword()));
    }

    /**
     * Test when the email and password are valid.
     */
    @Test
    public void testGoodUser() {
        assertTrue(db.checkRegisteredUser(good.getEmail(), good.getPassword()));
    }

    /**
     * What occurs after the test has passed.
     */
    @After
    public void tearDown() {
        db.unregisterUser(good);
        db.unregisterUser(locked);
        db.unregisterUser(nUser);
    }
}

File: app/src/test/java/edu/gatech/donatrix/testRegisteredUser.kt
package edu.gatech.donatrix

import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.Assert.*

import edu.gatech.donatrix.dao.Database
import edu.gatech.donatrix.model.User
import edu.gatech.donatrix.model.UserType

class testRegisteredUser {
    internal val db = Database.instance
    internal var good: User
    internal var locked: User
    internal var nUser: User
    internal var unregistered: User
    internal var badUser: User

    @Before
    fun setUp() {
        good = User("pjharris1998@gmail.com", "pharris36", "Parker Harris", false, UserType.USER)
        locked = User("pjharris1998@gmail.com", "password", "Pete Harris", true, UserType.USER)
        unregistered = User("blah@blah.com", "blah", "Blah", false, UserType.USER)
        nUser = User(null, "null", "Null", false, UserType.USER)
        badUser = User("pjharris1998@gmail.com", "password", "Parker Harris", false, UserType.USER)
        db!!.registerUser(good)
        db!!.registerUser(locked)
        db!!.registerUser(nUser)
    }

    @Test
    fun testUnregisteredUser() {
        assertFalse(db!!.checkRegisteredUser(unregistered.email, unregistered.password))
    }

    @Test
    fun testLockedUser() {
        assertFalse(db!!.checkRegisteredUser(locked.email, locked.password))
    }

    @Test
    fun testNullUser() {
        assertFalse(db!!.checkRegisteredUser(nUser.email, nUser.password))
    }

    @Test
    fun testBadPassword() {
        assertFalse(db!!.checkRegisteredUser(badUser.email, badUser.password))
    }

    @Test
    fun testGoodUser() {
        assertTrue(db!!.checkRegisteredUser(good.email, good.password))
    }

    @After
    fun tearDown() {
        db!!.unregisterUser(good)
        db!!.unregisterUser(locked)
        db!!.unregisterUser(nUser)
    }
}